compiler/rustc_mir_transform/src/coroutine.rs

```diff
@@ -1339,13 +1333,10 @@ fn create_coroutine_resume_function<'tcx>(
 
     make_coroutine_state_argument_indirect(tcx, body);
 
-    match transform.coroutine_kind {
```

Comment:
Change this back to a match and make it exhaustive, please.

`!matches` IMO makes it much easier to forget adding a variant in the future.

Reply:
Like this? It can be be even more exhaustive (i.e. on `Movability` and `CoroutineSource`)

Reply:
This is good enough